MetaMask/metamask-extension
# Jl/caip25 permission migration/move request grant hooks into mmc

## **Description**

* Adds `requestPermittedChainsPermission`, `requestPermittedChainsPermissionIncremental`, and `requestCaip25Permission` methods to `MetamaskController`
* Uses these new hooks in `wallet_requestPermissions`, `eth_requestAccounts`, `wallet_switchEthereumChain`, `wallet_addEthereumChain`
* Removes `wallet_requestPermissions` update caveat flow
* Make `wallet_requestPermissions` no longer support anything but `eth_requestAccounts` and `endowment:permittedChains`

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/29213?quickstart=1)

## **Related issues**

See: https://github.com/MetaMask/metamask-extension/pull/27847#discussion_r1881143169

## **Manual testing steps**

Only user facing change is that `wallet_requestPermissions` should correctly replace instead update existing permissions again

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
